### ISSUE TITLE
Typings

### ISSUE THREAD

**Issue by philjones88** [godaddy/kubernetes-client#131]
A set of typings for us Typescript users would be really cool.

**Comment by Yantrio** [comment]
I would also love this, what's the probability of it happening?

**Comment by gisenberg** [comment]
I'm currently working on this.

**Comment by channiganfsu** [comment]
What is the current state of this issue?

**Comment by gisenberg** [comment]
@channiganfsu @Yantrio I've got a first cut at support for TypeScript typings for kubernetes-client. I'd appreciate your review over here: https://github.com/godaddy/kubernetes-client/pull/174

**Comment by terrysdev9** [comment]
any chance that [#174](https://github.com/godaddy/kubernetes-client/pull/174) will merged into master soon? I see it's waiting on one more approval?

**Comment by gisenberg** [comment]
@terrysdev9 This change was included in the 3.18.0 release. Let me know how it goes!

**Comment by terrysdev9** [comment]
Sweet!!! Thanks @gisenberg !

**Comment by rmacfie** [comment]
As far as I can tell, the declaration file isn't included in the NPM package.

**Comment by gisenberg** [comment]
@rmacfie Fixed in kubernetes-client@3.18.1 - thanks for the heads up!

**Comment by alvis** [comment]
any work for 5.0? Now with the change in the new version, the type definitions are completely unusable.
